src/editors/collectionDocumentsProvider.ts

```diff
@@ -0,0 +1,43 @@
+import * as vscode from 'vscode';
+import { EJSON } from 'bson';
+
+import ConnectionController from '../connectionController';
+
+export const NAMESPACE_URI_IDENTIFIER = 'namespace';
+export const VIEW_COLLECTION_SCHEME = 'VIEW_COLLECTION_SCHEME';
+export const DOCUMENTS_LIMIT = 10;
+
+export default class CollectionViewProvider implements vscode.TextDocumentContentProvider {
+  _connectionController: ConnectionController;
+
+  constructor(connectionController: ConnectionController) {
+    this._connectionController = connectionController;
+  }
+
+  onDidChangeEmitter = new vscode.EventEmitter<vscode.Uri>();
+  onDidChange = this.onDidChangeEmitter.event;
+
+  provideTextDocumentContent(uri: vscode.Uri): Promise<string> {
+    return new Promise(resolve => {
+      const dataservice = this._connectionController.getActiveConnection();
+
+      // Remove the `namespace=` attribute from the uri query.
+      const namespace = uri.query.slice(NAMESPACE_URI_IDENTIFIER.length + 1);
```

Comment:
I'm working on the `code lens` implementation currently and I've changed this uri query parsing to use `URLSearchParams`: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams